### PR TITLE
feat: Allow to show logs of all logging types by proxying messages

### LINKED ISSUE #979: [Bug] Logreader warns about `log_type` to be `syslog`
### Steps to reproduce
1. configure 'log_type' to be 'syslog',
2. wait for warning

### Expected behaviour
This is one of the defined ways to log system messages, so everything should be fine

### Actual behaviour
Visible message:
`Information Logreader File-based logging must be enabled to access logs from the Web UI. Your `log_type` is currently set to: [syslog]. If you feel this is an error, please verify `log_type` in your config.php and check the Nextcloud Administration Manual. This is not an actual log entry.`

### PR DESCRIPTION
**Pull request by susnux**
* Resolves #979

This creates an internal log file for all non file loggers, this enables showing all kind of logs.
The internal log file is then rotated by a background job, I think rotation makes more sense to keep the behavior similar to the file logger log file.
Maybe we can add a feature in the future to read also rotated logs (Nextcloud & internal log).

### TODO
* [x] Log rotation / truncating is required otherwise the log would grow infinitely.
* [x] Adjust unit tests

### CODE REVIEW

**Comment by kesselb** [comment]
I like the idea :+1:

**Comment by come-nc** [comment]
@susnux Would it not be possible to access syslog logs without duplicating them?
Are there other backends than file and syslog?

**Comment by kesselb** [comment]
> Are there other backends than file and syslog?

- Errorlog (using php's error_log)
- Systemdlog (requires https://github.com/systemd/php-systemd / https://github.com/uboslinux/ubos-php, both seem unmaintained now, and therefore we can consider deprecating it)

**Comment by susnux** [comment]
> Would it not be possible to access syslog logs without duplicating them?

syslog in every case I know not allowed to be read by PHP, and I think this makes sense to not read other apps output.

Basically what this is doing is doing it similar as `tee`, so you have your native log for e.g. other logging solutions or journald integration, but also be able to have this in Nextcloud for quick analysis.

**Comment by come-nc** [comment]
> > Would it not be possible to access syslog logs without duplicating them?
> 
> syslog in every case I know not allowed to be read by PHP, and I think this makes sense to not read other apps output.
> 
> Basically what this is doing is doing it similar as `tee`, so you have your native log for e.g. other logging solutions or journald integration, but also be able to have this in Nextcloud for quick analysis.

But you are basically duplicating log, this is the same as supporting to have several log backend active at once, to have both files and syslog, no?

**Comment by MrRinkana** [comment]
> But you are basically duplicating log, this is the same as supporting to have several log backend active at once, to have both files and syslog, no?

However, supporting multiple log backends active at the same time has an important benefit: it makes the duplication explicit/intended (for example with multiple "log_type" statements in the config.php). 

We would really like the combination of logging to the usual nextcloud.log while also logging some desired events to a remote syslog (trough rsyslog filters; sending only bruteforce attempts to a central log server).

However if we in the future would decide to log everything to the remote syslog (maybe very high rate of logs, or usage of some fancy log reader tool) we would not want duplicate logs and would disable the logviewer app anyway - likely the same as people using just log_type syslog now. 

So basically I really vouch for supporting multiple log_type statements in the config!
